cypress-io/cypress
# chore: capture versions of relevant dependencies with `x-dependencies` header

- Closes https://github.com/cypress-io/cypress-internal/issues/598

### Additional details


On the **initial** call to the Cypress manifest, we want to include any relevant dependencies and their versions if they exist in the project using the `x-dependencies` header.

### Steps to test


1. Create a project that has some of the dependencies in [these lists](https://github.com/cypress-io/cypress/blob/1df282c75baa8245eb147339f47ec6324d98308c/packages/scaffold-config/src/dependencies.ts#L179-L246)
2. Open Cypress with `DEBUG="cypress:data-context:sources:VersionsDataSource"`
3. Verify that on the **initial** log that reads `retrieving latest version information with headers:`, the `x-dependencies` header is present and contains the correct dependencies and versions that exist in your project, none that don't exist in your project, and none that don't exist in the lists linked above.
4. Verify that on subsequent requests to the Cypress manifest endpoint, the `x-dependencies` header is **not**
 included in the list of headers.

### How has the user experience changed?


This should not change the experience for the user.

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
